File: solutions/csharp/binary-search/1/BinarySearch.cs
using System;

public static class BinarySearch
{
    public static int Find(int[] input, int value)
    {
        var lo = 0;
        var hi = input.Length;

        while (lo < hi)
        {
            var i = (lo + hi) / 2;
            var cmp = value - input[i];

            if (cmp > 0)
                lo = i + 1;
            else if (cmp < 0)
                hi = i;
            else
                return i;
        }

        return -1;
    }
}

File: solutions/go/binary-search/1/binary_search.go
package binarysearch

func SearchInts(list []int, key int) int {
    lo := 0
    hi := len(list)-1

    for lo <= hi {
        i := (lo + hi) / 2
       
        if list[i] < key {
            lo = i + 1
        } else if list[i] > key {
            hi = i - 1
        } else {
        	return i
        }
    }

    return -1
}
